frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -136,35 +144,97 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
-      ...(subscription?.status?.installedCSV
-        ? [
-            k8sKill(
-              ClusterServiceVersionModel,
-              {
-                metadata: {
-                  name: subscription.status.installedCSV,
-                  namespace: subscription.metadata.namespace,
+    const uninstallOperator = () => {
+      const operatorUninstallPromises = [
+        k8sKill(SubscriptionModel, subscription, {}, deleteOptions),
+        ...(subscription?.status?.installedCSV
+          ? [
+              k8sKill(
+                ClusterServiceVersionModel,
+                {
+                  metadata: {
+                    name: subscription.status.installedCSV,
+                    namespace: subscription.metadata.namespace,
+                  },
                 },
-              },
-              {},
-              deleteOptions,
-            ),
-          ]
-        : []),
-      ...(removePlugins
-        ? [k8sPatch(ConsoleOperatorConfigModel, consoleOperatorConfig, [patch])]
-        : []),
-    ];
-    const operandDeletionPromises = deleteOperands
-      ? operands.map((operand: K8sResourceCommon) => {
-          const model = modelFor(referenceFor(operand));
-          return k8sKill(model, operand, {}, deleteOptions);
+                {},
+                deleteOptions,
+              ),
+            ]
+          : []),
+        ...(removePlugins
+          ? [k8sPatch(ConsoleOperatorConfigModel, consoleOperatorConfig, [patch])]
+          : []),
+      ];
+
+      handleOperatorUninstallPromise(Promise.all(operatorUninstallPromises))
+        .then(() => {
+          setOperatorUninstallFinished(true);
         })
-      : [];
-    if (operandDeletionPromises.length > 0) {
+        .catch(() => {
+          setOperatorUninstallFinished(true);
+        });
+    };
+
+    const setOperandErrorsTo = (
+      operators: K8sResourceCommon[],
+      errorMsg: string,
+    ): OperandError[] => {
+      return operators.reduce((acc: OperandError[], curr, i) => {
+        return acc.concat({
+          operand: operands[i],
+          errorMessage: errorMsg,
+        });
+      }, []);
+    };
+
+    const verifyDeletionOfOperands = () => {
+      const url = `${window.SERVER_FLAGS.basePath}api/list-operands?name=${subscriptionName}&namespace=${subscriptionNamespace}`;
+      let operandVerificationErrors: OperandError[] = [];
+      let displayingVerificationResults = false;
+      let intervalId;
+
+      const finishVerification = () => {
+        clearInterval(intervalId);
+        setOperandsDeleteInProgress(false);
+        setOperandsDeleteFinished(true);
+        if (operandVerificationErrors.length === 0) {
+          uninstallOperator();
+        } else {
+          // show operand deletion verification errors on results page
+          setOperandDeletionErrors(operandVerificationErrors);
+          setOperatorUninstallFinished(true);
+        }
+      };
+
+      intervalId = setInterval(() => {
```

Comment:
I think this logic is more complicated than it needs to be. I'd recommend using a `useEffect` hook to handle things like polling. For example:

```ts
React.useEffect(() => {
  let intervalID;
  if (operandsDeleteInProgress) {
    intervalID = setInterval(() => {
      // ...
    }, 2000);
  }
  return () => {
    if (intervalID) {
      clearInterval(intervalID);
    }
  };
}, [operandsDeleteInProgress]);
```

Then you can be sure to clear the interval no matter what happens.

Reply:
Hi @spadgett, last commit has this change to `useEffect`.  However, the 'interval' does a lot of things (set errors, continue to uninstall, etc..) and thus has a lot of dependecies and caused a lot of refactoring.

The `useEffect` seems to start and finish/clear the interval many times?  Not sure why?  The clean-up function should only run during an 'unmount': 

"The clean-up function runs before the component is removed from the UI to prevent memory leaks. Additionally, if a component renders multiple times (as they typically do), the previous effect is cleaned up before executing the next effect." 

Here are some console outputs:
```
--> starting interval...
cleared interval!!!
--> starting interval...
deleting (daves-proj)-rhpam-trial
deleting daves-proj-link-rhpam-trial
in interval...
cleared interval!!!
--> starting interval...
(2) in interval...
cleared interval!!!
```
Still seems to work, though.  :thinking:

Reply:
It's doing that because you have `operands` in the dependency list. You don't necessarily want that. I think you have too much logic in that `useEffect` hook. It should just really be setting up the interval IMO, and then you could have another `useEffect` that handles changes to the operand count for example.

Reply:
Hi @spadgett, ok reduced that `useEffect` to just:
```
React.useEffect(() => {
    let intervalID;

    if (operandsDeleteInProgress) {
      intervalID = startVerificationInterval();
    }
    return () => {
      if (intervalID) {
        clearInterval(intervalID);
      }
    };
  }, [operandsDeleteInProgress, startVerificationInterval]);
```